### PR TITLE
Document serialization interface and what to test

### PR DESCRIPTION
**Pull request by bf4**
Extracted from #954

ref: https://github.com/rails-api/active_model_serializers/issues/878#issuecomment-96073030

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -322,6 +322,48 @@ class PostSerializer < ActiveModel::Serializer
...
+- `#read_attribute_for_serialization(name)`: gets the attribute value for serialization
+
+ActiveRecord::Base objects include ActiveModel::Serialization
+
+The additionally must implement:
+- `#as_json(options)`: Hash representation of a serialized resource, may delegate to #serializable_hash
+- `#to_json(options)`: string representation (JSON) to Hash from `as_json`
+- `#cache_key`: a (self-expiring) unique key for the instance, used by the adapter
+- `#id`: a unique identifier for the object
+- `::model_name`: an ActiveModel::Name instance, used by the serializer to identify the type
```
Comment (bf4) [inline README.md]: Done in #996

File: README.md [hunk 2]
```diff
@@ -322,6 +322,48 @@ class PostSerializer < ActiveModel::Serializer
...
+
+Serializable resources must include [`ActiveModel::Serialization`](https://github.com/rails/rails/blob/master/activemodel/lib/active_model/serialization.rb)
+or implement its interface:
+
+- `#serializable_hash(options)`: hash representation of a resources attributes
+- `#read_attribute_for_serialization(name)`: gets the attribute value for serialization
+
+ActiveRecord::Base objects include ActiveModel::Serialization
+
+The additionally must implement:
```
Comment (locks) [inline README.md]: `They`?

File: README.md [hunk 3]
```diff
@@ -322,6 +322,48 @@ class PostSerializer < ActiveModel::Serializer
...
+- `#to_json(options)`: string representation (JSON) to Hash from `as_json`
+- `#cache_key`: a (self-expiring) unique key for the instance, used by the adapter
+- `#id`: a unique identifier for the object
+- `::model_name`: an ActiveModel::Name instance, used by the serializer to identify the type
+
+ActiveRecord::Base objects implement these methods by default.
+
+## Manual Serialization
+
+For the simple case, you may serialize a object with the below snippet:
```
Comment (locks) [inline README.md]: `with the snippet` below sounds better IMO, or `with the following snippet`

**Comment by locks** [comment]
@bf4 hi :) left you some notes

**Comment by bf4** [comment]
@locks and they are awesome.  This PR is basically on hold while #994 is dished out.  I also intent to extract parts of it into linters / 'shared tests' such as in #996 

Wanna look at #954 (if you haven't already)?  It requires brains. :cloud: :bulb: :smile:

**Comment by locks** [comment]
Oof, my brain can't handle code right now, I'll review prose though! :smile:

File: README.md [hunk 4]
```diff
@@ -322,6 +322,48 @@ class PostSerializer < ActiveModel::Serializer
 end
 ```
 
+## Serializable Resource Interface
+
+Serializable resources must include [`ActiveModel::Serialization`](https://github.com/rails/rails/blob/master/activemodel/lib/active_model/serialization.rb)
+or implement its interface:
+
+- `#serializable_hash(options)`: hash representation of a resources attributes
```
Comment (beauby) [inline README.md]: Could we detail the current/future options?

File: README.md [hunk 5]
```diff
@@ -322,6 +322,48 @@ class PostSerializer < ActiveModel::Serializer
...
+
+## Manual Serialization
+
+For the simple case, you may serialize a object with the below snippet:
+
+```ruby
+serializer_class = ActiveModel::Serializer.serializer_for(obj)
+serializer = serializer_class.new(obj)
+adapter = ActiveModel::Serializer.adapter.new(serializer)
+adapter.serializable_hash
```
Comment (beauby) [inline README.md]: This is outdated.

Comment (bf4) [inline README.md]: Yeah, it's been needing an update for a few months.. I just haven't circle back to it

**Comment by joaomdmoura** [comment]
BTW not README anymore, focus on spread this information the docs. :smile:

**Comment by bf4** [comment]
I've been meaning to come back to this to add serializer test helpers. the linter and and interface docs are done

**Pull request closed without merging**